packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart

```diff
@@ -748,19 +748,25 @@ void main() {
     final Completer<WebViewController> controllerCompleter =
         Completer<WebViewController>();
 
+
+    final testUrl = 'data:text/html;charset=utf-8;base64,$getTitleTestBase64';
     await tester.pumpWidget(
       Directionality(
         textDirection: TextDirection.ltr,
         child: WebView(
-          initialUrl: 'data:text/html;charset=utf-8;base64,$getTitleTestBase64',
+          initialUrl: testUrl,
           onWebViewCreated: (WebViewController controller) {
             controllerCompleter.complete(controller);
           },
           onPageStarted: (String url) {
-            pageStarted.complete(null);
+            if (url == testUrl) {
```

Comment:
Can you add a comment for why this is needed? And maybe log when `url != testUrl` to confirm your speculative fix?

Reply:
Added a comment. I had thought about logging, but didn't want to since we generally don't want extra logging in test runs, and if this theory is right this would be permanent code.

I don't think logging would help us much with confirming since it was flaky; we'd never know if a pass without logging meant the theory was wrong, or that it's a run that wouldn't have failed in the first place. We'll know pretty soon whether it worked or not regardless, by whether the flaking stops.